packages/grafana-ui/src/components/uPlot/utils.ts

```diff
@@ -369,16 +369,30 @@ function hasNegSample(data: unknown[], samples = 50) {
     lastIdx--;
   }
 
+  let negCount = 0;
+  let zeroCount = 0;
+  let posCount = 0;
+
   if (lastIdx >= firstIdx) {
     const stride = Math.max(1, Math.floor((lastIdx - firstIdx + 1) / samples));
 
     for (let i = firstIdx; i <= lastIdx; i += stride) {
       const v = data[i];
 
-      if (v != null && (v < 0 || Object.is(v, -0))) {
-        return true;
+      if (v != null) {
+        if (v < 0 || Object.is(v, -0)) {
+          negCount++;
+        } else if (v > 0) {
+          posCount++;
+        } else if (v === 0) {
+          zeroCount++;
```

Comment:
lint says zero never used... do we need it?

Reply:
i guess not. if it's all zeros it will be positive.